### PR TITLE
change-request-part1

### PR DESCRIPTION
**Pull request by mazoorek**

### CODE REVIEW

File: lib/models/widget_model.dart [hunk 1]
```diff
@@ -89,6 +91,45 @@ class DashboardWidget {
...
+    }
+  }
+
+  bool hasWarningStatus(WidgetStatus widgetStatus) {
+    return widgetStatus == WidgetStatus.CHECKBOX_UNKNOWN ||
+        widgetStatus == WidgetStatus.UNKNOWN ||
+        widgetStatus == WidgetStatus.UNSTABLE;
+  }
+
+  bool isErrorWidget() {
```
Comment (szymon-owczarzak) [inline lib/models/widget_model.dart:110]: Please drop `Widget` from the method name and be more `DRY` so instead calling `widget.isErrorWidget()` we will call `widget.isError()` - please apply for similar cases as well.

File: lib/providers/filter_provider.dart [hunk 2]
```diff
@@ -36,47 +36,17 @@ class FilterProvider with ChangeNotifier {
   List<DashboardWidget> getFilteredWidgetList(List<DashboardWidget> widgetList) {
     return widgetList.where((widget) {
       if (this.isWarningFilterPresent && this.isErrorFilterPresent) {
-        return this.isWarningOrErrorWidget(widget);
+        return widget.isWarningOrErrorWidget();
```
Comment (szymon-owczarzak) [inline lib/providers/filter_provider.dart:39]: be `DRY`

File: lib/providers/filter_provider.dart [hunk 3]
```diff
@@ -36,47 +36,17 @@ class FilterProvider with ChangeNotifier {
...
       } else if (this.isWarningFilterPresent) {
-        return isWarningWidget(widget);
+        return widget.isWarningWidget();
```
Comment (szymon-owczarzak) [inline lib/providers/filter_provider.dart:41]: DRY

File: lib/providers/filter_provider.dart [hunk 4]
```diff
@@ -36,47 +36,17 @@ class FilterProvider with ChangeNotifier {
...
       } else if (this.isErrorFilterPresent) {
-        return isErrorWidget(widget);
+        return widget.isErrorWidget();
```
Comment (szymon-owczarzak) [inline lib/providers/filter_provider.dart:43]: DRY

File: lib/providers/filter_provider.dart [hunk 5]
```diff
@@ -36,47 +36,17 @@ class FilterProvider with ChangeNotifier {
...
-          EnumToString.fromString(WidgetStatus.values, widget.content[DashboardWidget.WIDGET_STATUS_KEY]);
-      return widgetStatus == WidgetStatus.CHECKBOX_FAIL ||
-          widgetStatus == WidgetStatus.ERROR_CONNECTION ||
-          widgetStatus == WidgetStatus.ERROR_CONFIGURATION ||
-          widgetStatus == WidgetStatus.ERROR ||
-          widgetStatus == WidgetStatus.FAIL;
-    } else {
-      return false;
-    }
-  }
```
Comment (szymon-owczarzak) [inline lib/providers/filter_provider.dart:78]: I like that change :)

**Review by szymon-owczarzak — CHANGES_REQUESTED** [review]
Please apply `DRY` (Don't repeat yourself) rule to code.

File: lib/providers/filter_provider.dart (hunk 5 above)
Comment (mazoorek) [reply lib/providers/filter_provider.dart:78]: ;)

File: lib/providers/filter_provider.dart (hunk 2 above)
Comment (mazoorek) [reply lib/providers/filter_provider.dart:39]: done

File: lib/providers/filter_provider.dart (hunk 3 above)
Comment (mazoorek) [reply lib/providers/filter_provider.dart]: done

File: lib/providers/filter_provider.dart (hunk 4 above)
Comment (mazoorek) [reply lib/providers/filter_provider.dart]: done

File: lib/models/widget_model.dart (hunk 1 above)
Comment (mazoorek) [reply lib/models/widget_model.dart]: done

File: lib/providers/filter_provider.dart (hunk 2 above)
Comment (mazoorek) [reply lib/providers/filter_provider.dart:39]: done

**Review by szymon-owczarzak — APPROVED** [review]

**Pull request merged**